pkg/server/registry/apigroups/acorn/apps/pull.go

```diff
@@ -50,3 +56,12 @@ func (s *PullAppImageStrategy) Create(ctx context.Context, obj types.Object) (ty
 func (s *PullAppImageStrategy) New() types.Object {
 	return &apiv1.AppPullImage{}
 }
+
+func (v PullAppImageNameValidator) ValidateName(ctx context.Context, obj runtime.Object) (result field.ErrorList) {
+	appPullImage := obj.(*apiv1.AppPullImage)
+	if len(strings.Split(appPullImage.Name, ".")) == 2 {
+		result = append(result, field.Invalid(field.NewPath("metadata", "name"), appPullImage.Name, "invalid name\nTo update a nested Acorn or a service, update the parent Acorn instead."))
```

Comment:
nit: I think "invalid name" is implied by the field path and error type, so we could drop it (and the new-line) from the description.

Reply:
:+1: updated